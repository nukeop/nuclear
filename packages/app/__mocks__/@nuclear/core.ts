const initialStoreState = () => ({
  equalizer: {
    selected: 'Default'
  },
  downloads: [],
  favorites: {
    albums: [],
    tracks: []
  },
  playlists: []
});

let mockStore = initialStoreState();

module.exports = {
  store: {
    get: (key: string) => mockStore[key] || {},
    set: (key: string, value: any) => {
      mockStore[key] = value;
    },
    clear: () => mockStore = initialStoreState()
  },
  createApi: () => ({
    app: {},
    store: {},
    React: jest.fn(),
    ReactDOM: jest.fn()
  }),
  getOption: () => '',
  rest: {
    LastFmApi: class {
      constructor() {}

      getTagInfo() {}
      getTagTracks() {}
      getTagAlbums() {}
      getTagArtists() {}
      getTopTags = () => Promise.resolve()
      getTopTracks = () => Promise.resolve()
    }
  },
  settingsConfig: jest.requireActual('@nuclear/core/src/settings').settingsConfig,
  SettingType: {
    BOOLEAN: 'boolean',
    LIST: 'list',
    NODE: 'node',
    NUMBER: 'number',
    STRING: 'string',
    DIRECTORY: 'directory'
  }
};
