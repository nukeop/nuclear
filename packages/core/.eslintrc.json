{
  "extends": [
    "../../.eslintrc.json"
  ],
  "env": {
    "node": true,
    "browser": true,
    "commonjs": true
  },
  "rules": {
    "node/no-missing-import": 0,
    "node/no-extraneous-import": 0
  }
}
