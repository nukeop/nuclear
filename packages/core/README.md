# `@nuclear/core`

> TODO: description

## Usage

```
const core = require('@nuclear/core');

// TODO: DEMONSTRATE API
```
