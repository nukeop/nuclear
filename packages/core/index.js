export {
  LastFmApi,
  isElectron
} from './lib';
