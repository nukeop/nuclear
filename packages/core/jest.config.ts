export default async () => ({
  transform: {
    '^.+\\.(ts|tsx|js|jsx)$': 'ts-jest'
  },
  moduleNameMapper: {
    '^react$': '<rootDir>/../../node_modules/react'
  }
});
