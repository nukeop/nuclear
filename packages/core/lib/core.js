'use strict';

module.exports = core;

function core() {
  // TODO
}
