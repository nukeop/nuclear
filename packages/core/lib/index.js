import 'isomorphic-fetch';
export { default as LastFmApi } from './rest/lastfm';
