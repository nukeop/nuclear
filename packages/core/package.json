{
  "name": "@nuclear/core",
  "version": "0.6.16",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/nukeop/nuclear.git"
  },
  "author": "nukeop <nukeop@gumblert.tech>",
  "license": "AGPL-3.0",
  "bugs": {
    "url": "https://github.com/nukeop/nuclear/issues"
  },
  "homepage": "https://github.com/nukeop/nuclear#readme",
  "main": "src/index.ts",
  "scripts": {
    "lint": "eslint \"{test,src}/**/**.{js,ts}\"",
    "lint:fix": "eslint \"{test,src}/**/**.{js,ts}\" --fix",
    "test": "jest",
    "coverage": "jest --coverage"
  },
  "dependencies": {
    "@babel/core": "^7.9.0",
    "@babel/preset-env": "^7.9.5",
    "@babel/preset-react": "^7.9.4",
    "babel-jest": "^26.6.3",
    "bandcamp-scraper": "1.4.1",
    "cheerio": "^1.0.0-rc.5",
    "deezer-public-api": "^1.0.4",
    "electron-store": "^5.1.0",
    "electron-timber": "^0.5.1",
    "get-artist-title": "^1.2.0",
    "get-lyrics-hd": "^0.0.1",
    "isomorphic-fetch": "^2.2.1",
    "jest": "^26.6.3",
    "md5": "^2.2.1",
    "megalodon": "^3.6.1",
    "nodebrainz": "^2.1.1",
    "react": "^16.14.0",
    "react-dom": "^16.14.0",
    "redux": "^4.0.5",
    "search-azlyrics": "0.0.3",
    "ts-jest": "^26.5.5",
    "youtube-playlist": "^1.0.2",
    "ytsr": "^3.5.0"
  },
  "devDependencies": {
    "@types/isomorphic-fetch": "0.0.35",
    "@types/jest": "^26.0.22",
    "@types/lodash": "^4.14.158",
    "@types/md5": "^2.1.33",
    "@types/react": "^16.9.43",
    "eslint": "^7.18.0",
    "mock-require": "^3.0.3",
    "ts-node": "^8.10.2",
    "typescript": "^4.2.4",
    "ytdl-core": "^4.8.3"
  }
}
