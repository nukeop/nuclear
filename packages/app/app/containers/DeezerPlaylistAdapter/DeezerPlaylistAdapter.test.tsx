/* eslint-disable @typescript-eslint/ban-ts-comment */
import { rest } from '@nuclear/core';

import { buildStoreState } from '../../../test/storeBuilders';
import { mountedComponentFactory, setupI18Next } from '../../../test/testUtils';

jest.mock('@nuclear/core/src/rest');

describe('Deezer playlist adapter', () => {
  beforeAll(() => {
    setupI18Next();

    // @ts-ignore
    rest.Deezer = {
      getEditorialCharts: jest.fn().mockResolvedValue({
        playlists: {
          data: [{
            id: 1,
            title: 'Playlist 1'
          }]
        }
      })
    };
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should display a Deezer playlist', async () => {
    const { component } = mountComponent();
    await component.findByTestId('playlist-view');
    expect(component.asFragment()).toMatchSnapshot();
  });

  const mountComponent = mountedComponentFactory(
    ['/editorial-playlist/1'],
    buildStoreState()
      .withDashboard()
      .withPlugins()
      .withConnectivity()
      .build()
  );
});
