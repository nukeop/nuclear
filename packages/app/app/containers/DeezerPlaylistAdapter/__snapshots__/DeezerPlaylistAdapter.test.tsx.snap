// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deezer playlist adapter should display a Deezer playlist 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="playlist_view_container"
      data-testid="playlist-view"
    >
      <div
        class="playlist"
      >
        <div
          class="playlist_view_info"
        >
          <div>
            <img
              class="playlist_thumbnail"
              src="test-file-stub"
            />
          </div>
          <div
            class="playlist_header"
          >
            <label
              class="playlist_header_label"
            >
              Playlist
            </label>
            <div
              class="playlist_name"
            >
              Playlist 1
            </div>
            <div
              class="playlist_details"
            >
              <span>
                2 tracks
              </span>
            </div>
            <div
              class="playlist_buttons"
            >
              <button
                class="ui pink circular button nuclear button play_button"
              >
                <i
                  aria-hidden="true"
                  class="play icon"
                />
                 Play
              </button>
              <button
                class="ui basic circular button nuclear button more_button"
                data-testid="more-button"
              >
                <i
                  aria-hidden="true"
                  class="ellipsis horizontal icon"
                />
              </button>
            </div>
          </div>
        </div>
        <table
          class="track_table"
          role="table"
        >
          <thead>
            <tr
              role="row"
            >
              <th
                colspan="1"
                role="columnheader"
                style="cursor: pointer;"
                title="Toggle SortBy"
              >
                <span
                  class="title_with_icon"
                  data-testid="position-header"
                >
                  <i
                    aria-hidden="true"
                    class="hashtag icon"
                  />
                  <i
                    aria-hidden="true"
                    class="sort content ascending icon"
                  />
                </span>
              </th>
              <th
                colspan="1"
                role="columnheader"
                style="cursor: pointer;"
                title="Toggle SortBy"
              >
                <span
                  class="center_aligned"
                >
                  <i
                    aria-hidden="true"
                    class="image icon"
                  />
                </span>
              </th>
              <th
                colspan="1"
                role="columnheader"
                style="cursor: pointer;"
                title="Toggle SortBy"
              >
                Â 
              </th>
              <th
                colspan="1"
                role="columnheader"
                style="cursor: pointer;"
                title="Toggle SortBy"
              >
                <span
                  class="title_with_icon"
                >
                  Title
                  <i
                    aria-hidden="true"
                    class="sort content ascending disabled icon"
                  />
                </span>
              </th>
              <th
                colspan="1"
                role="columnheader"
                style="cursor: pointer;"
                title="Toggle SortBy"
              >
                <span
                  class="title_with_icon"
                >
                  Artist
                  <i
                    aria-hidden="true"
                    class="sort content ascending disabled icon"
                  />
                </span>
              </th>
              <th
                colspan="1"
                role="columnheader"
              >
                <span
                  class="select_header"
                >
                  <div
                    class="ui fitted checkbox nuclear checkbox"
                    style="cursor: pointer;"
                  >
                    <input
                      class="hidden"
                      readonly=""
                      tabindex="0"
                      title="Toggle All Rows Selected"
                      type="checkbox"
                      value=""
                    />
                    <label />
                  </div>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            data-rbd-droppable-context-id="1"
            data-rbd-droppable-id="track_table"
            role="rowgroup"
          >
            <tr
              class=""
              data-rbd-draggable-context-id="1"
              data-rbd-draggable-id="track 1 0"
              data-testid="track-table-row"
              role="row"
            >
              <td
                class="position_cell narrow"
                data-testid="position-cell"
                role="cell"
              >
                <button
                  class="ui pink tiny circular icon button nuclear button add_button"
                  data-testid="play-now"
                >
                  <i
                    aria-hidden="true"
                    class="play icon"
                  />
                </button>
                <span
                  class="position_cell_value"
                >
                  1
                </span>
              </td>
              <td
                class="thumbnail_cell narrow"
                role="cell"
              >
                <img
                  class="thumbnail"
                  src="test-file-stub"
                />
              </td>
              <td
                class="favorite_cell narrow"
                role="cell"
              >
                <button
                  class="ui tiny basic circular icon button nuclear button borderless"
                >
                  <i
                    aria-hidden="true"
                    class="heart outline icon"
                  />
                </button>
              </td>
              <td
                class="title_cell"
                data-testid="title-cell"
                role="cell"
              >
                <span
                  class="title_cell_content"
                >
                  <span
                    class="title_cell_value"
                  >
                    track 1
                  </span>
                  <span
                    class="title_cell_buttons"
                  >
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="add-to-queue"
                    >
                      <i
                        aria-hidden="true"
                        class="plus icon"
                      />
                    </button>
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="track-popup-trigger"
                    >
                      <i
                        aria-hidden="true"
                        class="ellipsis horizontal icon"
                      />
                    </button>
                  </span>
                </span>
              </td>
              <td
                role="cell"
              >
                artist 1
              </td>
              <td
                class="select_cell narrow"
                role="cell"
              >
                <div
                  class="ui fitted checkbox nuclear checkbox"
                  style="cursor: pointer;"
                >
                  <input
                    class="hidden"
                    readonly=""
                    tabindex="0"
                    title="Toggle Row Selected"
                    type="checkbox"
                    value=""
                  />
                  <label />
                </div>
              </td>
            </tr>
            <tr
              class=""
              data-rbd-draggable-context-id="1"
              data-rbd-draggable-id="track 2 1"
              data-testid="track-table-row"
              role="row"
            >
              <td
                class="position_cell narrow"
                data-testid="position-cell"
                role="cell"
              >
                <button
                  class="ui pink tiny circular icon button nuclear button add_button"
                  data-testid="play-now"
                >
                  <i
                    aria-hidden="true"
                    class="play icon"
                  />
                </button>
                <span
                  class="position_cell_value"
                >
                  2
                </span>
              </td>
              <td
                class="thumbnail_cell narrow"
                role="cell"
              >
                <img
                  class="thumbnail"
                  src="test-file-stub"
                />
              </td>
              <td
                class="favorite_cell narrow"
                role="cell"
              >
                <button
                  class="ui tiny basic circular icon button nuclear button borderless"
                >
                  <i
                    aria-hidden="true"
                    class="heart outline icon"
                  />
                </button>
              </td>
              <td
                class="title_cell"
                data-testid="title-cell"
                role="cell"
              >
                <span
                  class="title_cell_content"
                >
                  <span
                    class="title_cell_value"
                  >
                    track 2
                  </span>
                  <span
                    class="title_cell_buttons"
                  >
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="add-to-queue"
                    >
                      <i
                        aria-hidden="true"
                        class="plus icon"
                      />
                    </button>
                    <button
                      class="ui tiny basic circular icon button nuclear button borderless title_cell_button"
                      data-testid="track-popup-trigger"
                    >
                      <i
                        aria-hidden="true"
                        class="ellipsis horizontal icon"
                      />
                    </button>
                  </span>
                </span>
              </td>
              <td
                role="cell"
              >
                artist 2
              </td>
              <td
                class="select_cell narrow"
                role="cell"
              >
                <div
                  class="ui fitted checkbox nuclear checkbox"
                  style="cursor: pointer;"
                >
                  <input
                    class="hidden"
                    readonly=""
                    tabindex="0"
                    title="Toggle Row Selected"
                    type="checkbox"
                    value=""
                  />
                  <label />
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
